api/src/main/java/org/openmrs/Cohort.java

```diff
@@ -115,11 +118,11 @@ public Cohort(String name, String description, Collection patientsOrIds) {
 		if (patientsOrIds != null) {
 			for (Object o : patientsOrIds) {
 				if (o instanceof Patient) {
-					memberIds.add(((Patient) o).getPatientId());
+					CohortMember cohortMember = new CohortMember(this, (Patient) o, new Date());
+					members.add(cohortMember);
 				} else if (o instanceof Integer) {
-					memberIds.add((Integer) o);
-				} else {
-					memberIds.add(Integer.valueOf(o.toString()));
+					CohortMember cohortMember = new CohortMember(this, new Patient((Integer) o), new Date());
+					members.add(cohortMember);
 				}
 			}
 		}
```

Comment:
This method should be deprecated, in theory every Constructor and method that takes in patient id(s) and an argument should be deprecated.